### DISCUSSION TITLE
Do Tables require a blank line to end?

### DISCUSSION THREAD

**Discussion by gabriel-check24** [micromark/micromark#160] [Q&A] [answered]
I tried parsing the [v8-docs](https://github.com/search?q=repo%3Av8%2Fv8.dev+%3A%3A%3Atable&type=code) with micromark and the GFM extension. The v8 repo includes this table wrapping syntax:

```
:::table-wrap
|table|demo|
|---|---|
|row|row|
:::
```

This results in `:::` being recognized as a token of type `tableRow`. It also generates HTML like that's part of the table: `<td>:::</td>`

```
import {micromark} from 'micromark'
import {gfm, gfmHtml} from 'micromark-extension-gfm'

const value = ':::table-wrap\n|table|demo|\n|---|---|\n|row|row|\n:::';

const result = micromark(value, {
  extensions: [gfm()],
  htmlExtensions: [gfmHtml()]
})

console.log(result)
```

This issue is not specific to tthree colons, `hello` would also be parsed as a row. Do tables require a blank line to end?

**Comment by wooorm** [comment, marked as answer]
> hello

Yes, hello is explicitly part of a table in that case.

```markdown
|x|
|-|
|y|
z
```

|x|
|-|
|y|
z

---

Whether the `:::` should be seen as a container instead of a cell, is unknown. There is no directive spec. I can see something for your ideas.

But generally, if you‘re writing markdown, always use blank lines between things.
